ironfish/src/primitives/target.ts

```diff
@@ -140,14 +141,31 @@ export class Target {
 
     // max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
     const diffInSeconds = (time.getTime() - previousBlockTimestamp.getTime()) / 1000
-    const sign = BigInt(Math.max(1 - Math.floor(diffInSeconds / 10), -99))
-    const offset = BigInt(previousBlockDifficulty) / BigInt(DIFFICULTY_ADJUSTMENT_DENOMINATOR)
 
     // diff = parent_diff + parent_diff / 2048 * max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
-    const difficulty = BigIntUtils.max(
-      BigInt(previousBlockDifficulty) + offset * sign,
-      Target.minDifficulty(),
-    )
+
+    const threshold = 1
+    const halfBucket = Math.floor(FREEZE_TIME_IN_SECONDS / 2)
+    const bucket =
```

Comment:
can we extract this to a method and test that? it's really weird to have things like `halfBucket` here.